### PR TITLE
Feat/check list items

### LINKED ISSUE #36: #16 list-detail-page: check list-items
## Value Proposition
As a user I would like to have the possibility to check off list entries so that I don't have to delete them but can use them again
## Description
user-story-16
## Acceptance Criteria
- [ ] list-items can be disabled, when clicked on the name
- [ ] onclick: displays the disabled list-item inside a disabled category
- [ ] disabled categories are listed below the active categories

## Tasks
- [x] create feature `feat/check-list-items`

**list-item**
- [ ] add an event handler (click) on the name of the list-item
- [ ] add a function that adds the same category (disabled one) of the list-item to the list and moves the list-item there

**disabled list-item**
- [ ] add an event handler (click) on the name of the list-item
- [ ] add a function that moves the list-item back to the active category and deletes the "deactivated" category
(if this category does not contain any deactivated list-items anymore)

### PR DESCRIPTION
**Pull request by philmetscher**
Heyho. I added the ability to check list-items, when you click on the name. These deactivated list-items then appear under the active categories in a deactivated category of the list-item.

(I also changed a lot of code for the `zustand` library)

### CODE REVIEW

File: components/category/Category.js [hunk 1]
```diff
@@ -25,52 +23,84 @@ export default function Category({ category, listId }) {
...
+        disabled={hasDisabledItems}
       >
         {category.name}
       </CategoryHeadline>
       {isExtended && (
         <DragDropContext onDragEnd={handleOnDragEnd}>
-          <Droppable droppableId={category.id}>
+          <Droppable
+            droppableId={
+              hasDisabledItems ? "disabled" + category.id : category.id
```
Comment (ThomasF85) [inline components/category/Category.js]: modifying the id here is pretty hacky and changes the semantic meaning and information the id contains. A better way would be to give your  updateListItemIndex function another (boolean) parameter, for example like this:
```
updateListItemIndex: (destination, source, onlyDisabledItems) 
```

File: components/category/Category.js [hunk 2]
```diff
@@ -25,52 +23,84 @@ export default function Category({ category, listId }) {
...
+                        id={listItem.id}
+                        index={index}
+                        checked={listItem.checked}
+                        listId={listId}
+                      >
+                        {listItem.name}
+                      </CategoryListItem>
+                    ))}
+                  </>
                 )}
```
Comment (ThomasF85) [inline components/category/Category.js]: This is an unnecessary ternary operator. You can just calculate the relevant list items like this:
```
listItems.filter(
    (listItem) => listItem.categoryId === category.id && listItem.disabled === hasDisabledItems
  )
```

File: pages/list/[id].js [hunk 3]
```diff
@@ -21,37 +21,50 @@ export default function List() {
...
-            filteredCategories.map((category) => (
+          {activeCategories &&
+            activeCategories.map((category) => (
+              <DynamicCategory
+                key={category.id}
+                category={category}
+                listId={id}
+              />
+            ))}
+          {inactiveCategories &&
```
Comment (ThomasF85) [inline pages/list/[id].js]: unnecessary logical and operators, since both variables used here are arrays and all arrays are truthy values always.

**Review by ThomasF85 — CHANGES_REQUESTED** [review]
Please have a look at my requested changes.

File: components/category/Category.js [hunk 4]
```diff
@@ -23,16 +23,13 @@ export default function Category({ category, listId, hasDisabledItems }) {
...
-  const activeListItems = listItems.filter(
-    (listItem) => listItem.categoryId === category.id && !listItem.disabled
-  );
-  //inactiveListItems
-  const inactiveListItems = listItems.filter(
-    (listItem) => listItem.categoryId === category.id && listItem.disabled
+  const filteredListItems = listItems.filter(
+    (listItem) =>
+      listItem.categoryId === category.id &&
+      listItem.disabled === hasDisabledItems
```
Comment (Mark-Krensel) [inline components/category/Category.js:36]: if disabled has only one of two strings, you may want to set it to true/false instead

File: pages/list/[id].js [hunk 5]
```diff
@@ -29,8 +29,8 @@ export default function List() {
     list = lists.find((list) => list.id === id);
   }
 
-  let activeCategories;
-  let inactiveCategories;
+  let activeCategories = undefined;
+  let inactiveCategories = undefined;
```
Comment (ThomasF85) [inline pages/list/[id].js]: You should use null if you want to set the abscence of a value on purpose.

**Review by ThomasF85 — APPROVED** [review]
Looks good!

**Review by Mark-Krensel — APPROVED** [review]
Only read the changes from your last commit and it looks good to me.
as there were changes requested before, maybe @ThomasF85 wants to have a look again.

**Pull request merged**